engine/src/main/java/com/datatorrent/stram/plan/logical/DefaultKryoStreamCodec.java

```diff
@@ -56,37 +51,29 @@ public DefaultKryoStreamCodec()
   @Override
   public Object fromByteArray(Slice fragment)
   {
+    final Input input = new Input(fragment.buffer, fragment.offset, fragment.length);
     try {
-      ByteArrayInputStream is = new ByteArrayInputStream(fragment.buffer, fragment.offset, fragment.length);
-      Input input = new Input(is);
-      Object returnObject = kryo.readClassAndObject(input);
-      is.close();
-      return returnObject;
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
+      return kryo.readClassAndObject(input);
+    } finally {
+      input.close();
     }
   }
 
   @Override
-  public Slice toByteArray(T info)
+  public Slice toByteArray(T o)
   {
-    Slice slice = null;
+    final Output output = new Output(32, -1);
     try {
-      ByteArrayOutputStream os = new ByteArrayOutputStream();
-      Output output = new Output(os);
```

Comment:
It is not a direct pass through. Output creates an internal byte array buffer and flushes it to the OutputStream (if one is provided) when the internal buffer becomes full. As in this case OutputStream is ByteArrayOutputStream anyway, this leads to 3 unnecessary byte arrays copy operations.